### PR TITLE
[device/celestica] Implement Watchdog APIs based on the new platform API 

### PR DESCRIPTION
**Pull request by mudsut4ke**
**- What I did**

- Implement Watchdog APIs.
  - arm()
  - disarm()
  - is_armed()
  - get_remaining_time()
- Implement get_reboot_cause function on Chassis APIs.

**- How I did it**
1. Implement Watchdog API based on sonic platform API base
2. Init Watchdog object in Chassis class.
3.  Implement get_reboot_cause function on Chassis class.

**- How to verify it**
- Create python script follow the new platform API design doc to verify it (Azure/SONiC#285.)

Signed-off-by: Wirut Getbamrung wgetbumr@celestica.com

### CODE REVIEW

**Comment by lguohan** [comment]
retest this please

File: device/celestica/x86_64-cel_e1031-r0/sonic_platform/chassis.py [hunk 1]
```diff
@@ -94,3 +106,32 @@ def install_component_firmware(self, component_name, image_path):
...
+
+        if sw_reboot_cause != "Unexpected reboot":
+            reboot_cause = self.REBOOT_CAUSE_NON_HARDWARE
+        elif hw_reboot_cause == "0x11":
+            reboot_cause = self.REBOOT_CAUSE_POWER_LOSS
+        elif hw_reboot_cause == "0x33" or hw_reboot_cause == "0x55":
+            reboot_cause = self.REBOOT_CAUSE_WATCHDOG
+        else:
+            reboot_cause = self.REBOOT_CAUSE_HARDWARE_OTHER
+            description = 'Unknow reason'
```
Comment (jleveque) [inline device/celestica/x86_64-cel_e1031-r0/sonic_platform/chassis.py]: Typo: "Unknown"

File: device/celestica/x86_64-cel_e1031-r0/sonic_platform/watchdog.py [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+        self.state_path = "/sys/class/watchdog/%s/state" % self.wdt_main_dev_name
+        self.timeout_path = "/sys/class/watchdog/%s/timeout" % self.wdt_main_dev_name
+        # Set default value
+        self._disable()
+        self.armed = False
+        self.timeout = self._gettimeout(self.timeout_path)
+
+    def _is_wd_main(self, dev):
+        """
+        Checks if dev is Mellanox main watchdog
```
Comment (jleveque) [inline device/celestica/x86_64-cel_e1031-r0/sonic_platform/watchdog.py]: Mellanox? Copy/paste error?

File: device/celestica/x86_64-cel_seastone-r0/sonic_platform/watchdog.py [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+        self.state_path = "/sys/class/watchdog/%s/state" % self.wdt_main_dev_name
+        self.timeout_path = "/sys/class/watchdog/%s/timeout" % self.wdt_main_dev_name
+        # Set default value
+        self._disable()
+        self.armed = False
+        self.timeout = self._gettimeout(self.timeout_path)
+
+    def _is_wd_main(self, dev):
+        """
+        Checks if dev is Mellanox main watchdog
```
Comment (jleveque) [inline device/celestica/x86_64-cel_seastone-r0/sonic_platform/watchdog.py]: Mellanox? Copy/paste error?

File: device/celestica/x86_64-cel_seastone-r0/sonic_platform/chassis.py [hunk 4]
```diff
@@ -94,3 +106,33 @@ def install_component_firmware(self, component_name, image_path):
...
+        if sw_reboot_cause != "Unexpected reboot":
+            reboot_cause = self.REBOOT_CAUSE_NON_HARDWARE
+            description = sw_reboot_cause
+        elif hw_reboot_cause == "0x11":
+            reboot_cause = self.REBOOT_CAUSE_POWER_LOSS
+        elif hw_reboot_cause == "0x22":
+            reboot_cause = self.REBOOT_CAUSE_WATCHDOG,
+        else:
+            reboot_cause = self.REBOOT_CAUSE_HARDWARE_OTHER
+            description = 'Unknow reason'
```
Comment (jleveque) [inline device/celestica/x86_64-cel_seastone-r0/sonic_platform/chassis.py]: Typo: "Unknown"

File: device/celestica/x86_64-cel_seastone-r0/sonic_platform/watchdog.py (hunk 3 above)
Comment (mudsut4ke) [reply device/celestica/x86_64-cel_seastone-r0/sonic_platform/watchdog.py]: It was my mistake

File: device/celestica/x86_64-cel_seastone-r0/sonic_platform/chassis.py (hunk 4 above)
Comment (mudsut4ke) [reply device/celestica/x86_64-cel_seastone-r0/sonic_platform/chassis.py]: Updated https://github.com/Azure/sonic-buildimage/pull/3123/commits/971021f0b122a445794938ac4c7dc1c89a30afdc

**Comment by mudsut4ke** [comment]
retest this please

**Pull request merged**